File: apps/backend/ModuAgent/core/interfaces/perception.py
from __future__ import annotations

from abc import ABC, abstractmethod
from typing import Any, Dict, Optional


class BasePerception(ABC):
    @abstractmethod
    def perceive(
        self,
        input_type: str,
        raw_content: bytes,
        language: Optional[str] = None,
        sensitivity_level: int = 0,
    ) -> Dict[str, Any]:
        pass


class BaseSensor(ABC):
    @abstractmethod
    def sensor_type(self) -> str:
        pass

    @abstractmethod
    def capture(self, context: Dict[str, Any]) -> bytes:
        pass

File: packages/modu-agent/src/core/interfaces/perception.ts
// 对应 Python: core/interfaces/perception.py
// BasePerception + BaseSensor 抽象接口

/**
 * 感知器抽象接口。
 * 对应 Python BasePerception（perceive）。
 */
export abstract class BasePerception {
  abstract perceive(
    inputType: string,
    rawContent: Uint8Array,
    language?: string | null,
    sensitivityLevel?: number,
  ): Promise<Record<string, any>> | Record<string, any>
}

/**
 * 传感器抽象接口。
 * 对应 Python BaseSensor（sensor_type / capture）。
 */
export abstract class BaseSensor {
  abstract sensorType(): string

  abstract capture(context: Record<string, any>): Promise<Uint8Array> | Uint8Array
}
